mii/entrypoints/api_server.py

```diff
@@ -0,0 +1,217 @@
+# Standard library imports
+import json
+import time
+import grpc
+import asyncio
+import argparse
+import threading
+from queue import Queue
+from typing import AsyncGenerator
+from concurrent.futures import ThreadPoolExecutor
+
+# Third-party imports
+import fastapi
+import uvicorn
+import mii
+from fastapi import FastAPI, Request
+from fastapi.exceptions import RequestValidationError
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import JSONResponse, StreamingResponse, Response
+from mii.grpc_related.proto.modelresponse_pb2_grpc import ModelResponseStub
+from mii.grpc_related.proto import modelresponse_pb2
+from mii.utils import kwarg_dict_to_proto
+
+# Local module imports
+from .data_models import CompletionRequest
+
+app = FastAPI()
+load_balancer = "localhost:50050"
+
+@app.post("/generate")
+async def generate(request: CompletionRequest) -> Response:
+    # TODO: Add support for multiple stop tokens, as for now only one is supported
+    # Check if stop token is a list
+    if request.stop is not None and isinstance(request.stop, list):
+        request.stop = request.stop[0]
+
+    # Set defaults
+    if request.max_tokens is None:
+        request.max_tokens = 128
+
+    if request.stream is None:
+        request.stream = False
+
+    if request.prompt is None:
+        return JSONResponse({"error": "Prompt is required."}, status_code=400)
+
+    if isinstance(request.prompt, str):
+        request.prompt = [request.prompt]
+
+    # Set up the generation arguments
+    generate_args = {
+        "ignore_eos": False,
+        "do_sample": True,
+        "return_full_text": False
+    }
+
+    # Set optional generation arguments
+    if request.max_length is not None:
+        generate_args["max_length"] = request.max_length
+    
+    if request.min_tokens is not None:
+        generate_args["min_new_tokens"] = request.min_tokens
+    
+    if request.max_tokens is not None:
+        generate_args["max_new_tokens"] = request.max_tokens
+    
+    if request.top_p is not None:
+        generate_args["top_p"] = request.top_p
+
+    if request.top_k is not None:
+        generate_args["top_k"] = request.top_k
+
+    if request.temperature is not None:
+        generate_args["temperature"] = request.temperature
+    
+    if request.stop is not None:
+        generate_args["stop"] = request.stop    
+
+    if request.stream:
+        generate_args["stream"] = True
+
+    channel = grpc.aio.insecure_channel(load_balancer)
+    stub = ModelResponseStub(channel)
+    requestData = modelresponse_pb2.MultiStringRequest(
+        request=request.prompt,
+        query_kwargs=kwarg_dict_to_proto(generate_args),
+    )
+
+    # Streaming case
+    if request.stream:
```

Comment:
Streaming is a bit broken right now. The outputs do not have spaces included (which is why we don't include this in any of our docs currently). We have a PR that I should be able to complete soon that will enable proper streaming support (#311). Until then, I think it would be best to not expose this incomplete feature here.

Reply:
Alright, should we just comment out or delete that part?

Reply:
Let's comment it out and raise an exception if streaming is true for right now. When I have the token streaming fixed, I will restore this functionality. Thank you!